docs/MA/viergewinnt6.cpp

```diff
@@ -0,0 +1,781 @@
+                std::sort(not_none_children.begin(), not_none_children.end(),[](const MinimaxNode& a, const MinimaxNode& b) {return a.value > b.value;});
+                //
+                sorted_children.insert(sorted_children.end(), not_none_children.begin(), not_none_children.end());
+                sorted_children.insert(sorted_children.end(), none_children.begin(), none_children.end());
+                this->children=sorted_children;
+                //
+                for (MinimaxNode& child : not_none_children) {child.sort(false);}
+            }
+            else if (!max_player) {
+                std::sort(not_none_children.begin(), not_none_children.end(),[](const MinimaxNode& a, const MinimaxNode& b) {return a.value < b.value;});
+                //
+                sorted_children.insert(sorted_children.end(), not_none_children.begin(), not_none_children.end());
+                sorted_children.insert(sorted_children.end(), none_children.begin(), none_children.end());
+                this->children=sorted_children;
+                //
+                for (MinimaxNode& child : not_none_children) {child.sort(true);}
+            }
+        }
+    }
+
+};
+
+class MinimaxPlayer {
+public:
+    MinimaxPlayer(int token, std::vector<std::vector<int>> board) : token(token), board(board) {
+        root_node.board = board;
+        root_node.player_am_zug = token;
+        root_node.token=token;
+        root_node.value_not_none = false;
+        root_node.value = 0;
+        root_node.depth = 0;
+        root_node.children = root_node.expand_node();
+        root_node.expanded=true;
+        //
+        minimax_counter=0;
+    }
+    MinimaxNode root_node;
+    int token;
+    std::vector<std::vector<int>> board;
+    int max_time=2;
+    int max_depth=10;
+    int starting_depth=1;
+
+    std::vector<std::vector<int>> minimaxer(const int depth, const std::chrono::duration<double> vergangene_zeit) {
+        auto start = std::chrono::high_resolution_clock::now();
+        //
+        std::vector<int> values;
+        std::vector<MinimaxNode> best_moves;
+        MinimaxNode best_move;
+        int best_value = -1000000;
+        std::vector<MinimaxNode>& root_node_children=root_node.children;
+        std::vector<std::vector<int>> return_board;
+        //
+        for (MinimaxNode& child : root_node_children){
+            int eval;
+            eval=child.minimax(-1000000,1000000,false, depth);
+            child.value=eval;
+            std::cout<<"a ";//child wurde fertig berechnet
+            //
+            auto now = std::chrono::high_resolution_clock::now();
+            std::chrono::duration<double> vergangene_zeit2 =(now+vergangene_zeit)-start;
+            if (vergangene_zeit2.count() >= max_time) {std::cout<<" NICHT FERTIG"; break;}
+        }
+        //
+        for (MinimaxNode& child : root_node_children) {values.push_back(child.value);}
+        for (int v : values) {if (v > best_value) {best_value = v;}}
```

Comment:
Müssten hier nicht die `best_moves` zurückgesetzt werden?

```suggestion
        for (int v : values) {
            if (v > best_value) {
                best_value = v;
                best_moves.clear();
            }
        }
```